### ISSUE TITLE
Linux: Core halts at startup (black screen) until a controller is detected

### ISSUE THREAD

**Issue by 7oxicshadow** [libretro/mupen64plus-libretro-nx#89]
This might happen on other OS's aswell but I can only test on Linux.

I think I have seen at least one other core do this but if you launch a rom from the command line without a controller plugged in the core just halts on a black screen with no prompts. If you then connect the controller it starts to run.

If this is by design, would it be possible to have a prompt on screen warning people that no controllers are detected?

**Comment by m4xw** [comment]
Can't think of anything in mupen blocking it, might be a frontend Issue.

**Comment by 7oxicshadow** [comment]
I just spent some time preparing a debug build when I noticed that even though it is halted on a black screen, if you leave it for about 30 seconds or more it does eventually start. More importantly the following comes up in the terminal window:

    [WARN] [SDL]: Failed to create rumble effect for joypad 0: Haptic: Error uploading effect to the device: Connection timed out.

I am starting to think that this is not a bug and more a problem associated with my setup. I use xboxdrv as the joypad driver and I think it is the cause of the delay. Once the timeout has occurred I can close retroarch and re-launch from the command line in the same manor and it boots the game straight away even though a joypad is not connected.

I will close this and should I find that it is a Retroarch / Core issue I will re-open.

Thanks.
